### PR TITLE
Set timeout for requests made by the EVP proxy

### PR DESCRIPTION
**Pull request by albertvaka**
### Motivation

Despite the trace-agent API server having a timeout (that defaults to 5 seconds), requests to the EVP proxy endpoint could continue running past this timeout because they were not cancelled themselves. In either case we would close the connection with the client without propagating the response from the server, but we would take way longer than intended.

### Describe how to test/QA your changes

Before the patch (and hardcoding the endpoint to a local python server that takes 8 seconds to respond):

```
$ time curl -v 127.0.0.1:8126/evp_proxy/v1/
*   Trying 127.0.0.1:8126...
* Connected to 127.0.0.1 (127.0.0.1) port 8126
> GET /evp_proxy/v1/ HTTP/1.1
> Host: 127.0.0.1:8126
> User-Agent: curl/8.4.0
> Accept: */*
> 
* Empty reply from server
* Closing connection
curl: (52) Empty reply from server

real	0m8.033s
user	0m0.006s
sys	0m0.010s
```

After the patch:

```
$ time curl -v 127.0.0.1:8126/evp_proxy/v1/
*   Trying 127.0.0.1:8126...
* Connected to 127.0.0.1 (127.0.0.1) port 8126
> GET /evp_proxy/v1/ HTTP/1.1
> Host: 127.0.0.1:8126
> User-Agent: curl/8.4.0
> Accept: */*
> 
* Empty reply from server
* Closing connection
curl: (52) Empty reply from server

real	0m5.033s
user	0m0.006s
sys	0m0.010s
```

Both close the connection without response, but the first one goes over the timeout.

### CODE REVIEW

File: tasks/go_deps.py [hunk 1]
```diff
@@ -180,7 +180,7 @@ def test_list(
                 filename = os.path.join(ctx.cwd, f"dependencies_{goos}_{goarch}.txt")
                 if not os.path.isfile(filename):
                     print(
-                        f"File {filename} does not exist. To execute the dependencies list check for the {binary} binary, please run the task `inv -e go-deps.generate --binaries {binary}"
```
Comment (pgimalac) [inline tasks/go_deps.py:183]: Nice catch !

**Review by pgimalac — APPROVED** [review]

**Comment by ajgajg1134** [comment]
Hello! Is it possible to add a small unit test verifying this change instead of needing to rely on manual testing using a separate python program? Thanks!

**Review by ajgajg1134 — APPROVED** [review]
looks good from APM! Thanks for adding a test, I think the unit test covers this pretty well so feel free to add a qa/done label to this PR so no manual QA jira ticket is created

**Review by knusbaum — APPROVED** [review]

**Comment by albertvaka** [comment]
@knusbaum I don't have permissions to merge this myself, but it looks like I can still use the merge queue flow though :)

**Pull request merged**